### ISSUE TITLE
Training cost of finetuning SpatialVLA on BridgeV2 data

### ISSUE THREAD

**Issue by OrangeSodahub** [SpatialVLA/SpatialVLA#70]
Hi, I have checked your paper and did not find the descriptions about training details of finetuing SpatialVLA. I would like to know that

1) the training cost (e.g., using 4 80G A100, how long it takes to finetuen SpatialVLA on BridgeV2 data) to reproduce your results.
2) the results in Table I and II, are from LoRA-finetuing or full-finetuing? I assume it is LoRA-finetuing, is that correct?
https://github.com/SpatialVLA/SpatialVLA?tab=readme-ov-file#-fine-tune-from-spatialvla

Thanks!

**Comment by OrangeSodahub** [comment]
cc @DelinQu Thanks!

**Comment by OrangeSodahub** [comment]
@DelinQu And I would like to double check the BridgeV2 data version is the one from oxe, right? Since you didn't explain how to prepare finetuing dataset here: https://github.com/SpatialVLA/SpatialVLA?tab=readme-ov-file#-fine-tune-from-spatialvla

Should we download finetuning (BridgeV2) data from here: https://docs.google.com/spreadsheets/d/1rPBD77tk60AEIGZrGSODwyyzs5FgCU9Uz3h-3_t2A9g/edit?gid=0#gid=0?

**Comment by DelinQu** [comment]
Hi [@OrangeSodahub](https://github.com/OrangeSodahub),

> 1. The training cost (e.g., using 4×80G A100, how long it takes to finetune SpatialVLA on BridgeV2 data).

If you finetune from [IPEC-COMMUNITY/spatialvla-4b-224-pt](https://huggingface.co/IPEC-COMMUNITY/spatialvla-4b-224-pt), it usually takes around 24 hours for 1–2 epochs on 4×A100 80G. In our setup, we used 8 GPUs, and the fine-tuning finished in under 12 hours.

> 2. Are the results in Table I and II from LoRA fine-tuning or full fine-tuning? I assume LoRA?

Tables I and II report results from full-parameter fine-tuning, which consistently performs better than LoRA-based tuning in our experiments.

> 3. Is the BridgeV2 data version the one from OXE? Since you didn’t specify dataset preparation details.

a. Download raw data from: [Google Sheet link](https://docs.google.com/spreadsheets/d/1rPBD77tk60AEIGZrGSODwyyzs5FgCU9Uz3h-3_t2A9g/edit?gid=0#gid=0)

b. Perform dataset filtering using [moojink/rlds_dataset_mod](https://github.com/moojink/rlds_dataset_mod)

Yes, the BridgeV2 dataset we used follows the OXE release.
You can also refer to this detailed comment for dataset preparation guidance:
👉 [openvla/openvla#101 (comment)](https://github.com/openvla/openvla/issues/101#issuecomment-2346640861)

**Comment by OrangeSodahub** [comment]
@DelinQu Thanks for your reply!

And I have two more follow-up questions:
1. I download raw Bridge data from google oxe, the folder is `bridge/0.1.0`, which is different from the default values in your `finetune_*.sh`: `mixture=bridge_orig` and `name=bridge_orig/1.0.0`, I'm so confused.
2. I notice that `data_root_dir=../open_x_embodiment_converted`, is that obtained by running the command from https://github.com/moojink/rlds_dataset_mod?tab=readme-ov-file#modifying-rlds-datasets on the raw oxe data? So what did you said "filtering" mean? this command just resize the image to 336 and convert to jpg.

**Comment by DelinQu** [comment]
Thanks for the follow-up questions!

1. Please refer to the official Bridge dataset at
https://rail.eecs.berkeley.edu/datasets/bridge_release/data/tfds/bridge_dataset/
The version we used is 1.0.0 (corresponding to bridge_orig/1.0.0 in our scripts).

2. We perform dataset filtering and conversion by running
[prepare_open_x.sh](https://github.com/moojink/rlds_dataset_mod/blob/main/prepare_open_x.sh)
from the [moojink/rlds_dataset_mod](https://github.com/moojink/rlds_dataset_mod) repository on the raw OXE data.
This process filters out low-quality trajectories and produces the final folder open_x_embodiment_converted.
Here, “filtering” not only resizes or converts images, but also removes incomplete or noisy episodes based on metadata (e.g., trajectory length, missing observations, etc.), ensuring the data quality for fine-tuning.

**Comment by OrangeSodahub** [comment]
@DelinQu Thanks for further explainations!

For 1, I just become confused again. You mentioned the BridgeV2 dataset you used is from OXE release (I download it and found it is `0.1.0`), and then you also said you used is `1.0.0` (from official bridge website). 🥲 

Another question is, your converted data is in `256×256`?

**Comment by DelinQu** [comment]
<img width="489" height="150" alt="Image" src="https://github.com/user-attachments/assets/ffe7ba65-218b-4bb4-8d6f-fd1e23091439" />

Apologies for the confusion! Let me clarify: the version I referred to is indeed from the official BridgeV2 website, which is v1.0.0 (you can find it here: https://rail-berkeley.github.io/bridgedata/). Due to some issues with downloading from OpenX, we manually obtained the latest v1.0.0 version.

Regarding the images, yes, we resized them to 256×256.

**Comment by OrangeSodahub** [comment]
@DelinQu Hi, I have run the lora finetuning in my machine, using 4×16=64 total batch size (4 L20 GPU), it needs ~800 hours, is that correct?

<img width="1740" height="384" alt="Image" src="https://github.com/user-attachments/assets/80674fce-f892-410a-9ef6-1e36f6e1dc10" />

Additionally, I found that the `accuracy` gradually decline from the inital 1, which seems to be not good.
